src/interface/data_collection.hpp

```diff
@@ -10,36 +10,49 @@
 // license in this material to reproduce, prepare derivative works, distribute copies to
 // the public, perform publicly and display publicly, and to permit others to do so.
 //========================================================================================
-#ifndef INTERFACE_CONTAINER_COLLECTION_HPP_
-#define INTERFACE_CONTAINER_COLLECTION_HPP_
+#ifndef INTERFACE_DATA_COLLECTION_HPP_
+#define INTERFACE_DATA_COLLECTION_HPP_
 
 #include <map>
 #include <memory>
+#include <stdexcept>
 #include <string>
 
-#include "interface/container.hpp"
-#include "interface/metadata.hpp"
-
 namespace parthenon {
+class Mesh;
 
 template <typename T>
-class ContainerCollection {
+class DataCollection {
  public:
-  ContainerCollection() {
-    containers_["base"] = std::make_shared<Container<T>>(); // always add "base" container
+  DataCollection() {
+    containers_["base"] = std::make_shared<T>(); // always add "base" container
+    pmy_mesh_ = nullptr;
   }
 
-  void Add(const std::string &label, const std::shared_ptr<Container<T>> &src);
+  void SetMeshPointer(Mesh *pmesh) { pmy_mesh_ = pmesh; }
+
+  std::shared_ptr<T> Add(const std::string &label, const std::shared_ptr<T> &src);
```

Comment:
Doesn't seem like this is implemented anywhere?

Reply:
It's in the source file https://github.com/lanl/parthenon/blob/51e2b8e593243e0bb1b3307a8e36699983586765/src/interface/data_collection.cpp#L25